hyrax.gemspec

```diff
@@ -26,6 +26,12 @@ EOF
   spec.version       = Hyrax::VERSION
   spec.license       = 'Apache2'
 
+  # Be pessimistic, rails does not follow sem-ver conventions and
+  # minor version releases can include breaking changes; see
+  # http://guides.rubyonrails.org/maintenance_policy.html
+  # https://github.com/projecthydra-labs/hyrax/issues/1028
+  spec.add_dependency 'rails', '>= 5.0.0', '< 5.2'
```

Comment:
After discussing this on stand-up yesterday, it sounds like the team was :-1: on including `'< 5.2'` in the Hyrax gemspec, because the cost of having to update the gemspec whenever Rails updates outweighs the benefit. I believe we should change the dependency to `'~> 5.0'` cc: @jcoyne @cbeer @atz